### DISCUSSION TITLE
[Support]: What cpu usage levels are considered troubling or concerning?

### DISCUSSION THREAD

**Discussion by Cold-Lemonade** [blakeblackshear/frigate#15135] [General Support]
### Describe the problem you are having

I have mini pc with Intel n100 cpu running Ubuntu 24.04 and 10 ip cameras. The status bar at the bottom of Frigate's GUI reports that my cpu percentage is generally around 45%. It did spike this morning when it was snowing heavily, which I assume is to be expected because there is more motion for Frigate to process.

My question is what level of cpu usage is considered troubling or concerning? In the mid-forty percent, Frigate's status bar still shows my cpu percentage in green so I assume it is fine. The Intel GPU is negligible and reported at zero percent. 

What are other Frigate users seeing as their cpu usage? Should I reduce the fps and bit rate on my cameras or add motion masks to lower the cpu usage? 

### Version

0.14.1-f4f3cfa

### What browser(s) are you using?

Microsoft Edge 131.0.2903.51 (Official build) (64-bit)

### Frigate config file

```yaml
NA
```

### Relevant Frigate log output

```shell
NA
```

### Relevant go2rtc log output

```shell
NA
```

### FFprobe output from your camera

```shell
NA
```

### Install method

Docker Compose

### docker-compose file or Docker CLI command

```yaml
NA
```

### Object Detector

Coral

### Network connection

Wired

### Camera make and model

Dahua and Dahua clones

**Comment by NickM-27** [comment]
There is nothing inherently problematic about the CPU usage. The main considerations are things like:
1. limiting power usage, the higher the CPU usage the higher the power usage, though on an n100 it is negligible regardless
2. headroom for future expansion, if you are running at 90% on average over longer periods of time for example then there is not much headroom to add more services
3. Temperatures, if you stay at a high CPU usage for too long and don't have adequate cooling then the CPU may throttle and this will potentially cause frames to be skipped in Frigate

**Reply by Cold-Lemonade** [reply]
Thank you @NickM-27. So the main concern would be if Frigate's detection process started skipping frames. I have never seen that reported on my set up.

The only other services that I plan on putting on this mini pc are an NTP server so the cameras' clocks can all sync and Home Assistant. I believe both of those require minimal resources.

I have a couple of cameras running at 30 fps for the high res stream. I'll try lowering this to 25 fps to see how that effects things. I doubt there will be any noticeable change in the quality of the recorded video to my eye but it may lower cpu usage meaningfully.

**Reply by Cold-Lemonade** [reply]
@NickM-27 -- I am now seeing a couple of skipped frames. Is there a file in the container that records the number of dropped frames over some time period? I only see it in the "system metrics" tab and was wondering if there any where that shows this information over a longer time period.

**Reply by NickM-27** [reply]
No that is the main area, having some skipped frames during periods of high activity is not really a big issue, just if it is continuous / very high number.

**Reply by NickM-27** [reply]
If you have home assistant you can enable the sensor there

**Reply by Cold-Lemonade** [reply]
Thank you. Would having a second Coral TPU aid in lowering the skipped frames? Or is the bottle neck the CPU?

**Comment by NickM-27** [comment]
Need to see a screenshot of the system metrics and camera metrics

**Reply by Cold-Lemonade** [reply]
uploaded them in a new comment below...

**Comment by Cold-Lemonade** [comment]
Here they are...

![metrics1](https://github.com/user-attachments/assets/a62dbbef-7143-4e11-90f0-60bb94fad45c)

![metrics2](https://github.com/user-attachments/assets/5a09b71c-22d2-43f5-9dd7-1f9f43192263)

![metrics3](https://github.com/user-attachments/assets/1424a764-0b29-4c39-b45e-df7c3cc8f95b)

![metrics4](https://github.com/user-attachments/assets/b6619441-d0ab-4298-b8a2-99430383c681)

![metrics5](https://github.com/user-attachments/assets/bd4be0a8-90c7-43c7-beed-d85ee0483715)

**Reply by NickM-27** [reply]
Not seeing any skipped here but yeah other coral would likely help the issue

**Reply by Cold-Lemonade** [reply]
Thanks. I've added a usb Coral and now my skipped frames are zero ever since.

**Reply by Cold-Lemonade** [reply]
@NickM-27 -- a quick follow-up question: Is the "CPU XX%" in the Frigate status bar measured for one core or for the actual cpu on the system. For example, I'm running the Intel N100 and my status bar says "CPU 39%". Does this mean that I'm using less than half of 1 core or almost 2 of my 4 cores?

**Reply by NickM-27** [reply]
The status bar shows overall usage on the CPU

**Reply by bracklanna** [reply]
Could you clarify that to make it absolutely clear?

Let's say it's a 4-core CPU, if it says "CPU 39%" at the bottom of the Frigate metrics window, is that 39% of an overall grand-total capacity of 100% for all 4 cores?

Or (as is sometimes the case on linux systems,) does that refer to 39% of 1 core, in which case the maximum possible system total CPU usage would be 400%?

**Reply by NickM-27** [reply]
The bottom bar is normalized across all cores, so that is total
